torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -1558,6 +1732,60 @@ const Expr* TermExpander::mutate(const Polynomial* v) {
   return lastNode;
 }
 
+const Expr* TermExpander::mutate(const MaxTerm* v) {
+  const auto& variables = v->variables();
+  if (variables.empty()) {
+    if (!v->scalar()) {
+      // This case should never happen because MaxTerm will be created only
+      // on valid Max expressions.
+      throw std::logic_error("empty maxterm op");
+    }
+    return v->scalar();
+  }
+  auto lastNodeHash = v->hasher().hash(variables[0]);
+  const Expr* max;
+  if (v->scalar()) {
+    max = new Max(variables[0], v->scalar(), v->propagate_nans());
+  } else {
+    max = variables[0];
+  }
+  for (size_t i = 1; i < variables.size(); i++) {
+    auto currNodeHash = v->hasher().hash(variables[i]);
+    if (currNodeHash != lastNodeHash) {
```

Comment:
Feels much better to unique these when you add them to the OpTerm, not when you expand them.

Reply:
Sure, good point. Done.